File: hw5/laptopSearch.php
<!DOCTYPE html>
<html>
<body>

<form class="search" action="laptopSearch.php" method="post">
    <input type="text" placeholder="Minimum speed" name="speed">
    <input type="text" placeholder="Mimimum RAM" name="ram">
    <input type="text" placeholder="Minimum HD size" name="hd">
    <input type="text" placeholder="Minimum Screen" name="screen">
    <button type="submit"><i class="fa fa-search"></i></button>
</form>

<?php

class TableRows extends RecursiveIteratorIterator
{
    function __construct($it)
    {
        parent::__construct($it, self::LEAVES_ONLY);
    }

    function current()
    {
        return "<td style='width: 150px; border: 1px solid black;'>" . parent::current() . "</td>";
    }

    function beginChildren()
    {
        echo "<tr>";
    }

    function endChildren()
    {
        echo "</tr>" . "\n";
    }
}

$servername = "localhost";
$username = "root";
$password = "password";
$dbname = "store";


$speed = $_POST['speed'];
$ram = $_POST['ram'];
$hd = $_POST['hd'];
$screen = $_POST['screen'];



try {
        $conn = new PDO("mysql:host=$servername;port=3306;dbname=$dbname", $username, $password);
        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
        $stmt = $conn->prepare("SELECT * FROM laptop, product WHERE speed >= '%" . $speed . "%' AND 
      ram >= '%" . $ram . "%' AND hd >= '%" . $hd . "%' AND screen >= '%" . $screen . "%' AND laptop.model=product.model");
        $count = $conn->query("SELECT count(*) FROM (SELECT * FROM laptop, product WHERE speed >= '%" . $speed . "%' AND 
      ram >= '%" . $ram . "%' AND hd >= '%" . $hd . "%' AND screen >= '%" . $screen . "%' AND laptop.model=product.model) as T")->fetchColumn();
        if ($count > 0) {

            echo "<div style='padding-left:16px; padding-bottom: 16px; padding-right: 16px'>
                        <table style='border: solid 1px black;'>
                </div>";


            $stmt->execute();

            // set the resulting array to associative
            $result = $stmt->setFetchMode(PDO::FETCH_ASSOC);

            foreach (new TableRows(new RecursiveArrayIterator($stmt->fetchAll())) as $k => $v) {
                echo $v;
            }

            $conn = null;
            echo "</table>";
        }
    } catch (PDOException $e) {
        echo "Error: " . $e->getMessage();
    }
    ?>
</body>
</html>

File: hw5/laptopSearch.java
import java.sql.*;
import java.util.Scanner;



public class laptopSearch {

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        System.out.print("Minimum Speed: ");
        Double speed = scanner.nextDouble();
        System.out.print("Minimum RAM: ");
        Double ram = scanner.nextDouble();
        System.out.print("Minimum HD size: ");
        Double hd = scanner.nextDouble();
        System.out.print("Minimum Screen size: ");
        Double screen = scanner.nextDouble();

        Connection con = null;
        PreparedStatement statement = null;
        ResultSet rs = null;

        try {
            con = DriverManager.getConnection("jdbc:mysql://localhost/3306/store" + "user=root&password=password");
            statement = con.prepareStatement("SELECT * FROM laptop, product WHERE speed >= ? AND ram >= ? AND hd >= ? AND screen >= ? AND laptop.model=product.model");
            statement.setDouble(1,speed);
            statement.setDouble(2,ram);
            statement.setDouble(3,hd);
            statement.setDouble(4,screen);
            rs = statement.executeQuery();
            ResultSetMetaData rsmd = rs.getMetaData();
            int columns = rsmd.getColumnCount();
            while (rs.next()){
                for (int i = 1; i < columns; i++){
                    System.out.println(rs.getString(i) + "\t");
                }
                System.out.println();
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }


    }
}
